modules/custom/az_news/az_news_feeds/src/EventSubscriber/AzNewsFeedsMigrateSubscriber.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\az_news_feeds\EventSubscriber;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\migrate\Event\EventBase;
+use Drupal\migrate\Event\MigrateEvents;
+use Drupal\migrate\Event\MigrateImportEvent;
+use Drupal\migrate_plus\Entity\Migration;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Modify the default migrate config with user input values.
+ */
+class AzNewsFeedsMigrateSubscriber implements EventSubscriberInterface {
+
+  /**
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Constructs a AzNewsFeedsMigrateSubscriber object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The factory for configuration objects.
+   */
+  public function __construct(
+    ConfigFactoryInterface $config_factory
+    ) {
+    $this->configFactory = $config_factory;
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('config.factory'),
+    );
+  }
```

Comment:
The `create()` method is only needed for classes implementing the ContainerInjection interface (Controllers, Forms, etc).  This class has to have its constructor arguments defined in a `services.yml` file.
```suggestion
```